packages/server/lib/config.ts

```diff
@@ -64,14 +63,29 @@ const convertRelativeToAbsolutePaths = (projectRoot, obj, defaults = {}) => {
   , {})
 }
 
-const validateNoBreakingConfig = (cfg) => {
-  return _.each(breakingOptions, ({ name, errorKey, newName, isWarning }) => {
-    if (_.has(cfg, name)) {
+const validateNoBreakingConfig = ({ config, configFile }) => {
```

Comment:
The [resolved configFile from option](https://github.com/cypress-io/cypress/blob/issue-18684/packages/server/lib/config.ts#L251) should be available on config. Should be able to pull like:
```
(cfg) = > {
const { configFile } = cfg
```